src/exchange/exchange.py

```diff
@@ -51,8 +52,10 @@ def replace(self, **kwargs: Dict[str, Any]) -> "Exchange":
         """Make a copy of the exchange, replacing any passed arguments"""
         if kwargs.get("messages") is None:
             kwargs["messages"] = deepcopy(self.messages)
-        if kwargs.get("checkpoints") is None:
-            kwargs["checkpoints"] = deepcopy(self.checkpoints)
+        if kwargs.get("checkpoint_data") is None:
```

Comment:
This breaks backwards compatibility, specifically breaking some uses in Goose: https://github.com/search?q=repo%3Asquare%2Fgoose%20checkpoints%3D&type=code. We'll need to update those references and anywhere else they might be (custom plugins repo's).

Reply:
Good catch. Will update once this PR gets merged in. We'll a major version bump to indicate the breaking changes, even though they are minor breaking changes.